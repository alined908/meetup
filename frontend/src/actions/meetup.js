import {GET_MEETUPS_REQUEST, GET_MEETUPS_SUCCESS, GET_MEETUPS_ERROR, 
    ADD_MEETUP, ADD_MEETUP_MEMBER, ADD_EVENT_OPTION, ADD_GLOBAL_MESSAGE, DELETE_MEETUP, EDIT_MEETUP,
    VOTE_MEETUP_EVENT, GET_MEETUP_EVENTS, DELETE_MEETUP_EVENT, ADD_MEETUP_EVENT, EDIT_MEETUP_EVENT} from "../constants/action-types";
import {axiosClient} from '../accounts/axiosClient';
import {history} from '../components/MeetupApp'

export const getMeetups = (data) => async dispatch => {
    dispatch({type: GET_MEETUPS_REQUEST})
    var params;
    if (data.type === "private"){
        params = {type: data.type, ...data.categories && {categories: data.categories}}
    } else {
        params = {
            type: data.type, ...data.categories && {categories: data.categories}, 
            latitude: data.coords.latitude, longitude: data.coords.longitude, 
            radius: data.coords.radius
        }
    }
    
    try {
        const response = await axiosClient.request({
            method: "GET",
            url: "/api/meetups/", 
            headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
            },
            params: params
        })
        setTimeout(() => dispatch({type: GET_MEETUPS_SUCCESS, payload: response.data.meetups}), 100)
    } catch(e){
        console.log(e)
        dispatch({type: GET_MEETUPS_ERROR, payload: e})
    }
}

export const getMeetup = (uri) => async dispatch => {
    try {
        const response = await axiosClient.get(
            `/api/meetups/${uri}/`, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
            }}
        )
        // console.log(response)
        dispatch({type: ADD_MEETUP, payload: response.data})
    } catch(e){
        history.push('/404')
    }
}

export const addMeetup = (formProps, redirectOnSuccess) => async dispatch => {
    var params = {...formProps, 
        date: formProps.date.getFullYear()+ "-" + ("0"+(formProps.date.getMonth()+1)).slice(-2) + "-" + ("0" + formProps.date.getDate()).slice(-2)
    }
    try {
        const response = await axiosClient.post(
            `/api/meetups/`, params, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: ADD_MEETUP, payload: response.data.meetup})
        console.log(response.data)
        redirectOnSuccess(response.data.meetup.uri)
    } catch(e){
        console.log(e)
    }
} 

export const editMeetup = (formProps, uri, redirectOnSuccess) => async dispatch => {
    console.log('edit meetup')
    var date = new Date(formProps.date)
    var params = {...formProps, 
        date: date.getFullYear()+ "-" + ("0"+(date.getMonth()+1)).slice(-2) + "-" + ("0" + date.getDate()).slice(-2)
    }
    try {
        const response = await axiosClient.patch(
            `/api/meetups/${uri}/`, params, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: EDIT_MEETUP, payload: response.data})
        console.log(response.data)
        redirectOnSuccess(uri)
    } catch(e){
        console.log(e)
    }
}

export const deleteMeetup = (uri) => async dispatch => {
    try {
        const response = await axiosClient.delete(
            `/api/meetups/${uri}/`, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        console.log(response)
        dispatch({type: DELETE_MEETUP, payload: uri})
        history.push(`/meetups`)
    } catch(e) {
        console.log(e)
    }
}

export const getMeetupEvents = (uri) => async dispatch => {
    try {
        const response = await axiosClient.get(
            `/api/meetups/${uri}/events/`, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        console.log(response)
        dispatch({type: GET_MEETUP_EVENTS, payload: {uri: uri, data: response.data}})
    } catch(e) {
        console.log(e)
    }
}

export const addMeetupEvent = (event) => async dispatch => {
    dispatch({type: ADD_MEETUP_EVENT, payload: event.message})
}

export const deleteMeetupEvent = (event) => async dispatch => {
    dispatch({type: DELETE_MEETUP_EVENT, payload: event.message})
}

export const reloadMeetupEvent = (event) => async dispatch => {
    dispatch({type: EDIT_MEETUP_EVENT, payload: event.message})
}

export const voteMeetupEvent = (event) => async dispatch => {
    dispatch({type: VOTE_MEETUP_EVENT, payload: event.message})
}
    
export const decideMeetupEvent = (event) => async dispatch => {
    dispatch({type: EDIT_MEETUP_EVENT, payload: event.message})
}

export const addMeetupMember = (event) => async dispatch => {
    dispatch({type: ADD_MEETUP_MEMBER, payload: event.message})
}

export const addEventOption = (event) => async dispatch => {
    dispatch({type: ADD_EVENT_OPTION, payload: event.message})
}

export const sendMeetupEmails = (uri) => async dispatch => {

    try {
        await axiosClient.post(
            `/api/meetups/${uri}/email/`, {}, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: ADD_GLOBAL_MESSAGE, payload: {type: "success", message: "Successfully sent emails"}})
    } catch(e) {
        dispatch({type: ADD_GLOBAL_MESSAGE, payload: {type: "error", message: "Unable to send emails"}})
    }
}