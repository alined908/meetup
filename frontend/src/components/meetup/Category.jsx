import React from 'react'
import {Button} from "@material-ui/core"

const Category = (props) => {
    const [variant, setVariant] = React.useState(props.clicked)

    const handleClick = () => {
        setVariant(variant === 'outlined' ? 'contained' : 'outlined')
    }

    return (
        <Button variant={variant} color="primary" onClick={() => handleClick()}>{props.category.label}</Button>
    )
}

export default Category