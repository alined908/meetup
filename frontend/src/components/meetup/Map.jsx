import React, {Component} from "react"
import ReactMapGL, {Marker, NavigationControl, FlyToInterpolator} from "react-map-gl"
import LocationOnIcon from '@material-ui/icons/LocationOn';
import * as d3 from 'd3-ease'

const token = "pk.eyJ1IjoiZGFuaWVsOTA4OTA4IiwiYSI6ImNrNnR6aTdtbTA0bmwzbW1ya3FncjdlcGkifQ.cBAvCpS5zDNwzyb_GUZaoA"

class Map extends Component {

    constructor(props) {
        super(props);
        this.state = {
          viewport: {
            latitude: this.props.location[0],
            longitude: this.props.location[1],
            zoom: 12.5,
            bearing: 0,
            pitch: 0
          }
        };
    }

    onLoad = () => {
        const viewport = {
            ...this.state.viewport,
            zoom: 15,
            transitionDuration: 6000,
            transitionInterpolator: new FlyToInterpolator(),
            transitionEasing: d3.easeSin
        }
        this.setState({viewport})
    }


    render() {
        return (
          <ReactMapGL
            {...this.state.viewport}
            width="450px"
            height="350px"
            mapStyle="mapbox://styles/mapbox/light-v9"
            onViewportChange={viewport => this.setState({viewport})}
            onLoad={this.onLoad}
            mapboxApiAccessToken={token}
          >
            <div style={{position: 'absolute', right: 25, top: 10}}>
                <NavigationControl showCompass={false}/>
            </div>
            <Marker latitude={this.props.location[0]} longitude={this.props.location[1]}>
                <LocationOnIcon fontSize="large" color="secondary"></LocationOnIcon>
            </Marker>
          </ReactMapGL>
        );
      }
}

export default Map