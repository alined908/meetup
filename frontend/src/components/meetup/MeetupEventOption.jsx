import React, { Component } from "react";
import { connect } from "react-redux";
import { voteStatus } from "../../constants/default-states";
import {
  ThumbUpOutlined as ThumbUpOutlinedIcon,
  ThumbDownOutlined as ThumbDownOutlinedIcon,
  ThumbUp as ThumbUpIcon,
  ThumbDown as ThumbDownIcon,
  Cancel as CancelIcon,
  CancelOutlined as CancelOutlinedIcon,
  Room as RoomIcon,
  Phone as PhoneIcon,
  Launch as LaunchIcon,
  Close as CloseIcon,
  ZoomIn as ZoomInIcon,
  MoreVert as MoreVertIcon
} from "@material-ui/icons";
import RestaurantPreview from "../restaurant/RestaurantPreview";
import { Link } from "react-router-dom";
import { ADD_GLOBAL_MESSAGE } from "../../constants/action-types";
import { Rating } from "../components";
import { Avatar, Tooltip, IconButton, Menu, MenuItem , ListItemIcon, Typography } from "@material-ui/core";
import PropTypes from "prop-types";
import {
  meetupEventOptionPropType,
  meetupMemberPropType,
  userPropType,
} from "../../constants/prop-types";
import styles from "../../styles/meetup.module.css";

class MeetupEventOption extends Component {
  constructor(props) {
    super(props);
    this.state = {
      preview: false,
      anchor: null,
      hover: false
    };
  }

  determineNumVotes = () => {
    let scores = { 1: 0, 2: 0, 3: 0 };
    for (let key in this.props.option.votes) {
      var vote = this.props.option.votes[key];
      scores[vote.status] += 1;
    }
    return scores;
  };

  determineStatus = () => {
    const user = this.props.user.id;
    if (user in this.props.option.votes) {
      return this.props.option.votes[user].status;
    }
    return 0;
  };

  determineClicked = (status, option) => {
    return status === option;
  };

  handleClick = (status) => {
    //If vote is ban and option vote is not ban and option is not banned and already used ban
    if (!this.props.isUserMember || this.props.isPast) {
      return;
    }
    let check = true;

    if (
      status === voteStatus.ban &&
      !this.props.option.banned &&
      this.props.members[this.props.user.id].ban
    ) {
      this.props.dispatch({
        type: ADD_GLOBAL_MESSAGE,
        payload: { type: "error", message: "Already used ban" },
      });
      check = false;
    }
    if (
      this.props.option.banned &&
      (this.props.user.id in this.props.option.votes
        ? this.props.option.votes[this.props.user.id].status !== voteStatus.ban
        : true)
    ) {
      this.props.dispatch({
        type: ADD_GLOBAL_MESSAGE,
        payload: { type: "error", message: "Cant vote on banned option" },
      });
      check = false;
    }

    if (check) {
      this.props.socket.voteMeetupEvent({
        user: this.props.user.id,
        option: this.props.option.id,
        status: status,
        meetup: this.props.meetup,
      });
    }
  };

  handleDeleteOption = () => {
    if (window.confirm("Are you sure you want to delete")) {
      const option = this.props.option.id;
      this.props.socket.deleteEventOption({
        user: this.props.user.id,
        option: option,
        meetup: this.props.meetup,
        event: this.props.event,
      });
    }
  };

  handleMenuClick = (e) => {
    this.setState({anchor: e.currentTarget})
  }

  handleMenuClose  = () => {
    this.setState({anchor: null})
  }

  handlePreview = () => {
    this.setState({ preview: !this.state.preview });
  };

  renderActions = (status, scores, banned) => {
    const [like, dislike, ban] = [
      this.determineClicked(status, voteStatus.like),
      this.determineClicked(status, voteStatus.dislike),
      this.determineClicked(status, voteStatus.ban),
    ];
    const checkOwn = (status) => {
      if (
        this.props.user.id in this.props.option.votes &&
        this.props.option.votes[this.props.user.id].status === status
      ) {
        return styles.myclick;
      } else {
        return "";
      }
    };

    return (
      <div className={styles.rstActions}>
        <div className={`${styles.rstActionIcon} ${checkOwn(voteStatus.like)}`}>
          {!like ? (
            <Tooltip title="Like">
              <ThumbUpOutlinedIcon
                disabled={banned}
                className={styles.clickable}
                onClick={() => this.handleClick(voteStatus.like)}
              />
            </Tooltip>
          ) : (
            <Tooltip title="Undo Like">
              <ThumbUpIcon
                disabled={banned}
                className={styles.clickable}
                color="primary"
                onClick={() => this.handleClick(voteStatus.like)}
              />
            </Tooltip>
          )}
          <span className={styles.rstActionScore}>
              {scores[1]}
          </span>
        </div>
        <div
          className={`${styles.rstActionIcon} ${checkOwn(voteStatus.dislike)}`}
        >
          {!dislike ? (
            <Tooltip title="Dislike">
              <ThumbDownOutlinedIcon
                disabled={banned}
                className={styles.clickable}
                onClick={() => this.handleClick(voteStatus.dislike)}
              />
            </Tooltip>
          ) : (
            <Tooltip title="Undo Dislike">
              <ThumbDownIcon
                disabled={banned}
                className={styles.clickable}
                onClick={() => this.handleClick(voteStatus.dislike)}
              />
            </Tooltip>
          )}
          <span className={styles.rstActionScore}>
              {scores[2]}
          </span>
        </div>
        <div className={`${styles.rstActionIcon} ${checkOwn(voteStatus.ban)}`}>
          {!banned ? (
            <Tooltip title="Ban">
              <CancelOutlinedIcon
                className={styles.clickable}
                onClick={() => this.handleClick(voteStatus.ban)}
              />
            </Tooltip>
          ) : (
            <Tooltip title="Undo Ban">
              <CancelIcon
                disabled={!ban}
                className={styles.clickable}
                color="secondary"
                onClick={() => this.handleClick(voteStatus.ban)}
              />
            </Tooltip>
          )}
          <span className={styles.rstActionScore}>
              {scores[3]}
          </span>
        </div>
      </div>
    );
  };

  renderRestauraunt = (data, isNotChosen) => {
    const banned = this.props.option.banned;
    const scores = this.determineNumVotes();
    const status = this.determineStatus();

    return (
      <>
        <div 
            className={styles.rst} 
            onMouseEnter={() => this.setState({hover: true})} 
            onMouseLeave={() => this.setState({hover: false})}
        >
          <Link to={`/restaurants/${data.url}`}>
            <img className={styles.rstImg} src={data.yelp_image} />
          </Link>
        </div>
        <div className={styles.rstCardBottom}>
          <div className={styles.rstInfo}>
            <div className={styles.rstInfoTop}>
                <Rating rating={data.rating} />
                <Tooltip title="Go To Restaurant Page" placement="top">
                    <Link to={`/restaurants/${data.url}`} className={styles.rstInfoName}>
                      <Typography variant="h6" noWrap>
                        {data.name}
                      </Typography>
                    </Link>
                </Tooltip>
            </div>  
            <div>
                <IconButton style={{color: "rgba(10,10,10, .95)"}} edge="end" onClick={this.handleMenuClick}>
                    <MoreVertIcon/>
                </IconButton>
                <Menu 
                    anchorEl={this.state.anchor} 
                    open={this.state.anchor} 
                    onClose={this.handleMenuClose}
                >
                    <MenuItem onClick={(e) => {this.handlePreview(); this.handleMenuClose(e);}}>
                        <ListItemIcon>
                            <ZoomInIcon color="primary" fontSize="small" />
                        </ListItemIcon>
                        <Typography variant="body2" noWrap>
                            Preview Option
                        </Typography>
                    </MenuItem>
                    <a
                        target="_blank"
                        rel="noopener noreferrer"
                        href={data.yelp_url}
                    >
                        <MenuItem onClick={this.handleMenuClose}>
                            
                            <ListItemIcon>
                                <LaunchIcon color="primary" fontSize="small" />
                            </ListItemIcon>
                            <Typography variant="body2" noWrap>
                                Go to Yelp Page
                            </Typography>
                            
                        </MenuItem>
                    </a>
                    {(isNotChosen && this.props.isUserMember) &&
                        <MenuItem onClick={(e) => {this.handleDeleteOption(); this.handleMenuClose(e);}}>
                            <ListItemIcon>
                                <CloseIcon color="secondary" fontSize="small" />
                            </ListItemIcon>
                            <Typography variant="body2" noWrap>
                                Delete Option
                            </Typography>
                        </MenuItem>
                    }
                </Menu>
            </div>
          </div>
          <div className={styles.rstCategories}>
            {data.price} &#8226;
            {data.categories.map((rc) => (
              <div key={rc.category.id} className={styles.categoryChip}>
                <Avatar
                  style={{ width: 20, height: 20 }}
                  variant="square"
                  src={`${process.env.REACT_APP_S3_STATIC_URL}${rc.category.api_label}.png`}
                >
                  <img
                    style={{ width: 20, height: 20 }}
                    alt={"&#9787;"}
                    src={`https://meetup-static.s3-us-west-1.amazonaws.com/static/general/panda.png`}
                  />
                </Avatar>
                {rc.category.label}
              </div>
            ))}
          </div>
          {this.props.full && this.renderActions(status, scores, banned)}
        </div>
      </>
    );
  };

  render() {
    const data = this.props.option.restaurant;
    const banned = this.props.option.banned;

    return (
      <>
        {this.props.full ? 
          <div
            className={`${styles.rstWrapper} ${
              banned ? styles.banned : ""
            } ${this.state.hover ? "elevate-2" : "elevate"}`}
            id={`option-${this.props.optionId}`}
          >
            {this.renderRestauraunt(data, this.props.full)}
          </div>
          :
          <div className={`${styles.rstHorz} ${this.state.hover ? "elevate-2" : "elevate"}`} id={`option-${this.props.optionId}`}>
            {this.renderRestauraunt(data, this.props.full)}
            <div className={styles.restaurantHorzInfo}>
              <div className={styles.rstHorzInfoEntry}>
                <RoomIcon /> <span>{data.location}</span>
              </div>
              <div className={styles.rstHorzInfoEntry}>
                <PhoneIcon /> <span>{data.phone}</span>
              </div>
            </div>
          </div>
        }
        {this.state.preview && (
          <RestaurantPreview
            handleClose={this.handlePreview}
            identifier={data.identifier}
          />
        )}
      </>
    )
  }
}

MeetupEventOption.propTypes = {
  socket: PropTypes.object.isRequired,
  isUserMember: PropTypes.bool.isRequired,
  event: PropTypes.number.isRequired,
  meetup: PropTypes.string.isRequired,
  full: PropTypes.bool.isRequired,
  data: PropTypes.object.isRequired,
  option: meetupEventOptionPropType,
  members: PropTypes.objectOf(meetupMemberPropType),
  user: userPropType,
};

function mapStateToProps(state, props) {
  return {
    option:
      state.meetup.meetups[props.meetup].events[props.event].options[
        props.optionId
      ],
    members: state.meetup.meetups[props.meetup].members,
    user: state.user.user,
  };
}

export default connect(mapStateToProps)(MeetupEventOption);
export { MeetupEventOption as UnderlyingMeetupEventOption };
