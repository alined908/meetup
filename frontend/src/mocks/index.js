import moment from 'moment'

//Category
export const category = {
    id: 1, 
    image: null,
    label: "Dessert", 
    api_label: "desserts"
}

//User
export const user = {
    id: 1, 
    email: "daniel@gmail.com", 
    first_name: "Daniel", 
    last_name: "Lee",
    avatar: null
}
export const user2 = {
    id: 2, 
    email: "example2@gmail.com", 
    first_name: "Bob", 
    last_name: "Jim", 
    avatar: null
}
export const user3 = {
    id: 3, 
    email: "example3@gmail.com", 
    first_name: "Bob", 
    last_name: "Jim", 
    avatar: null
}
export const settings = {
    radius: 25, 
    location: null, 
    latitude: null,
    longitude: null
}

export const userWithSettings = {
    ...user,
    settings: {...settings}
}
export const friend = {
    id: 1, 
    user: user2, 
    created_at: moment().format(), 
    chat_room: "something"
}
export const friends = [friend]
export const preference = {
    id: 1, 
    user, 
    category, 
    name: "Dessert", 
    ranking: 1, 
    timestamp: moment().format()
}
export const preferences = [preference]
export const friendInvite = {
    id: 1, 
    timestamp: moment().format(), 
    status: 1, 
    uri: "random", 
    sender: user, 
    receiver: user2
}

//Chat
export const message = {
    id: 1, 
    message: "hello", 
    timestamp: moment().format(), 
    room_id: 1, 
    sender: user, 
    is_notif: false
}
export const message2 = {
    id: 2, 
    message: "hello", 
    timestamp: moment().format(), 
    room_id: 1, 
    sender: user, 
    is_notif: true
}
export const messages = [message, message2]
export const members = {1: user, 2: user2}
export const room = {
    uri: {
        id: 1, 
        uri: "abc", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: 1, 
        meetup:null, 
        notifs: 0
    }
}
export const rooms = {
    friend: {
        id: 1, 
        uri: "abc", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: 1, 
        meetup:null, 
        notifs: 0
    },
    meetup: {
        id: 2, 
        uri: "xyz", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: null, 
        meetup:1, 
        notifs: 3
    }
}

//General
export const error = {
    message: "Something went wrong."
}

//Meetup
export const meetupMember1 = {
    1: {
        id: 1,
        user,
        ban: false,
        admin: true
    }
}

export const meetupMember2 = {
    2: {
        id: 2,
        user: user2,
        ban: false,
        admin: false
    }
}

export const meetupMember3 = {
    3: {
        id: 3,
        user: user3,
        ban: false,
        admin: false
    }
}

export const meetupMembers = {...meetupMember1, ...meetupMember2}

export const meetup = {
    uri: {
        id: 1, 
        name: "Meetup", 
        uri: "uri", 
        creator: user, 
        location: "location", 
        date: moment().format("YYYY-MM-DD"),
        members: meetupMembers,
        notifs: 0,
        public: true,
        categories: [category],
        latitude: 34.228754, 
        longitude: -118.2351192,
        events: {},
        isMeetupEventsInitialized : true
    }
}

export const meetups = {...meetup}

export const event = {
    1: {
        id: 1,
        meetup: 1,
        creator: user,
        title: "Dessert",
        start: moment().format(),
        end: moment().format(),
        chosen: null,
        categories: [category],
        options: {},
        price: "1, 2",
        distance: 16000,
        entries: {},
        random: true
    }
}
export const events = {...event}

export const option = {
    1 : {
        id: 1,
        score: 0,
        option: null,
        restaurant: {},
        votes: {},
        banned: false
    }
}

export const options = {...option}

export const tokens = {
    access: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNTg4MjI5OTEzLCJqdGkiOiIwOTg2YTg0MDA1ZGM0OGQ1ODEzYmIyNzE1OWViNGUyZCIsInVzZXJfaWQiOjEsInVzZXIiOnsiaWQiOjEsImVtYWlsIjoiZGFuaWVsQGdtYWlsLmNvbSIsImZpcnN0X25hbWUiOiJEYW5pZWwiLCJsYXN0X25hbWUiOiJMZWUiLCJhdmF0YXIiOm51bGwsInNldHRpbmdzIjp7InJhZGl1cyI6MjUsImxvY2F0aW9uIjpudWxsLCJsYXRpdHVkZSI6bnVsbCwibG9uZ2l0dWRlIjpudWxsfX19.FbffRthOKX5F0TGvqA5mZXuXUWNkUojzKexDGCoP518",
    refresh : "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoicmVmcmVzaCIsImV4cCI6MTU4ODM4Mjk0NiwianRpIjoiN2ZhNGY2MDE4OWM3NDA5ZmFjOTMwYmE4ZTRlMWFhMTMiLCJ1c2VyX2lkIjoxLCJ1c2VyIjp7ImlkIjoxLCJlbWFpbCI6ImRhbmllbEBnbWFpbC5jb20iLCJmaXJzdF9uYW1lIjoiRGFuaWVsIiwibGFzdF9uYW1lIjoiTGVlIiwiYXZhdGFyIjpudWxsLCJzZXR0aW5ncyI6eyJyYWRpdXMiOjI1LCJsb2NhdGlvbiI6bnVsbCwibGF0aXR1ZGUiOm51bGwsImxvbmdpdHVkZSI6bnVsbH19fQ.qqaY4Ye7rmyOKLbDS2EKz4SGWH0fwZ0MvrGP6NJ3IRE"
}