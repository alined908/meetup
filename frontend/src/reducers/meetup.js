import {GET_MEETUPS, ADD_MEETUP, DELETE_MEETUP, CLEAR_STORE, VOTE_MEETUP_EVENT, ADD_MEETUP_EVENT, GET_MEETUP_EVENTS, DELETE_MEETUP_EVENT, EDIT_MEETUP_EVENT} from '../constants/action-types';

const defaultState = {
    meetups: {},
    activeMeetup: null,
    isMeetupsInitialized: false,
    isMeetupEventsInitialized: false,
}

export default function meetupReducer(state = defaultState, action){
    switch (action.type){
        case GET_MEETUPS:
            return {...state, meetups: action.payload, isMeetupsInitialized: true}
        case ADD_MEETUP:
            return {...state, meetups: {...state.meetups, [action.payload.uri]: action.payload}}
        case DELETE_MEETUP:
            var meetups = {}
            Object.keys(state.meetups).map((key) => {
                if (key !== action.payload){
                    meetups[key] = state.meetups[key]
                }
            })
            return {...state, meetups: meetups}
        case GET_MEETUP_EVENTS:
           var meetups = {}
           const data = action.payload.data
           Object.keys(state.meetups).map((key) => 
               (key === action.payload.uri) ? meetups[key] = ({...state.meetups[key], events: data}) : (meetups[key] = state.meetups[key])
           )
            return {...state, meetups: meetups, isMeetupEventsInitialized: true}
        case ADD_MEETUP_EVENT:
            return {...state, 
                        meetups: {...state.meetups, 
                            [action.payload.meetup]: {...state.meetups[action.payload.meetup], 
                                events: {...state.meetups[action.payload.meetup].events, 
                                    ...action.payload.event}}}}
        case EDIT_MEETUP_EVENT:
            const event_id = action.payload.event_id
            return {...state, 
                        meetups: {...state.meetups, 
                            [action.payload.meetup]: {...state.meetups[action.payload.meetup],
                                events: {...state.meetups[action.payload.meetup].events, [event_id]: action.payload.event}}}}
        case VOTE_MEETUP_EVENT:
            var meetup = action.payload.meetup
            return {...state, meetups: {...state.meetups, 
                            [meetup]: {...state.meetups[meetup], 
                                events: {...state.meetups[meetup].events, 
                                    [action.payload.event]: {...state.meetups[meetup].events[action.payload.event], 
                                        options: {...state.meetups[meetup].events[action.payload.event].options, 
                                            [action.payload.option_id]: action.payload.option
                    }}}}}}
        case DELETE_MEETUP_EVENT:
            var events = {}
            const uri = action.payload.uri
            Object.keys(state.meetups[uri].events).map((key) => {
                if (key !== action.payload.event.toString()){
                    events[key] = state.meetups[uri].events[key]
                }
            })
            return {...state, meetups: {...state.meetups, [uri] : {...state.meetups[uri], events: events}}}
        case CLEAR_STORE:
            return defaultState
        default:
            return state
    }
}