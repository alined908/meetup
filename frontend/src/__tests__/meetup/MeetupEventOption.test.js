import React from "react";
import { shallow } from "enzyme";
import { UnderlyingMeetupEventOption } from "../../components/meetup/MeetupEventOption";
import * as mocks from "../../mocks";

const props = {
  socket: {},
  isUserMember: true,
  event: 1,
  meetup: "uri",
  full: true,
  option: mocks.option[1],
  data: mocks.option[1].id,
  members: mocks.meetup.members,
  user: mocks.user,
};

describe("MeetupWrapper tests", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render without crashing", () => {
    const wrapper = shallow(<UnderlyingMeetupEventOption {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
