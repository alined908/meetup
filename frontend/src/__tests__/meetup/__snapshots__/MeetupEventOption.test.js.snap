// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetupWrapper tests should render without crashing 1`] = `
<Fragment>
  <div
    className="center"
  >
    <div
      className="rstWrapper  elevate"
    >
      <div
        className="rst"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <Link
          to="/restaurants/undefined"
        >
          <img
            className="rstImg"
          />
        </Link>
      </div>
      <div
        className="rstCardBottom"
      >
        <div
          className="rstInfo"
        >
          <div
            className="rstInfoTop"
          >
            <Rating />
            <WithStyles(ForwardRef(Tooltip))
              placement="top"
              title="Go To Restaurant Page"
            >
              <Link
                className="rstInfoName"
                to="/restaurants/undefined"
              >
                <WithStyles(ForwardRef(Typography))
                  noWrap={true}
                  variant="h6"
                />
              </Link>
            </WithStyles(ForwardRef(Tooltip))>
          </div>
          <div>
            <WithStyles(ForwardRef(IconButton))
              edge="end"
              onClick={[Function]}
              style={
                Object {
                  "color": "rgba(10,10,10, .95)",
                }
              }
            >
              <MoreVertIcon />
            </WithStyles(ForwardRef(IconButton))>
            <WithStyles(ForwardRef(Menu))
              anchorEl={null}
              onClose={[Function]}
              open={null}
            >
              <WithStyles(ForwardRef(MenuItem))
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(ListItemIcon))>
                  <ZoomInIcon
                    color="primary"
                    fontSize="small"
                  />
                </WithStyles(ForwardRef(ListItemIcon))>
                <WithStyles(ForwardRef(Typography))
                  noWrap={true}
                  variant="body2"
                >
                  Preview Option
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(MenuItem))>
              <a
                rel="noopener noreferrer"
                target="_blank"
              >
                <WithStyles(ForwardRef(MenuItem))
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(ListItemIcon))>
                    <LaunchIcon
                      color="primary"
                      fontSize="small"
                    />
                  </WithStyles(ForwardRef(ListItemIcon))>
                  <WithStyles(ForwardRef(Typography))
                    noWrap={true}
                    variant="body2"
                  >
                    Go to Yelp Page
                  </WithStyles(ForwardRef(Typography))>
                </WithStyles(ForwardRef(MenuItem))>
              </a>
              <WithStyles(ForwardRef(MenuItem))
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(ListItemIcon))>
                  <CloseIcon
                    color="secondary"
                    fontSize="small"
                  />
                </WithStyles(ForwardRef(ListItemIcon))>
                <WithStyles(ForwardRef(Typography))
                  noWrap={true}
                  variant="body2"
                >
                  Delete Option
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(MenuItem))>
            </WithStyles(ForwardRef(Menu))>
          </div>
        </div>
        <div
          className="rstCategories"
        >
          $
           â€¢
          <div
            className="categoryChip"
            key="1"
          >
            <WithStyles(ForwardRef(Avatar))
              src="undefineddesserts.png"
              style={
                Object {
                  "height": 20,
                  "width": 20,
                }
              }
              variant="square"
            >
              <img
                alt="&#9787;"
                src="https://meetup-static.s3-us-west-1.amazonaws.com/static/general/panda.png"
                style={
                  Object {
                    "height": 20,
                    "width": 20,
                  }
                }
              />
            </WithStyles(ForwardRef(Avatar))>
            Dessert
          </div>
        </div>
        <div
          className="rstActions"
        >
          <div
            className="rstActionIcon "
          >
            <WithStyles(ForwardRef(Tooltip))
              title="Like"
            >
              <ThumbUpOutlinedIcon
                className="clickable"
                disabled={false}
                onClick={[Function]}
              />
            </WithStyles(ForwardRef(Tooltip))>
            <span
              className="rstActionScore"
            >
              0
            </span>
          </div>
          <div
            className="rstActionIcon "
          >
            <WithStyles(ForwardRef(Tooltip))
              title="Dislike"
            >
              <ThumbDownOutlinedIcon
                className="clickable"
                disabled={false}
                onClick={[Function]}
              />
            </WithStyles(ForwardRef(Tooltip))>
            <span
              className="rstActionScore"
            >
              0
            </span>
          </div>
          <div
            className="rstActionIcon "
          >
            <WithStyles(ForwardRef(Tooltip))
              title="Ban"
            >
              <CancelOutlinedIcon
                className="clickable"
                onClick={[Function]}
              />
            </WithStyles(ForwardRef(Tooltip))>
            <span
              className="rstActionScore"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
